analyses/cms-open-data-ttbar/analysis.py

```diff
@@ -147,22 +147,29 @@ def define_trijet_mass(df: ROOT.RDataFrame) -> ROOT.RDataFrame:
 
     # Build trijet combinations
     df = df.Define("Trijet", "ROOT::VecOps::Combinations(Jet_pt[Jet_pt_mask],3)")
-    df = df.Define("nTrijet", "Trijet[0].size()")
 
-    # Assign four-momentums to each trijet combination
+    
+    # Trijet_btag is a helpful array mask indicating whether or not the maximum btag value in Trijet is larger than the 0.5 threshold
     df = df.Define(
-        "Trijet_p4",
-        """
-        ROOT::RVec<ROOT::Math::PxPyPzMVector> Trijet_p4(nTrijet);
-        for (int i = 0; i < nTrijet; ++i)
-        {
-            int j1 = Trijet[0][i];
-            int j2 = Trijet[1][i];
-            int j3 = Trijet[2][i];
-            Trijet_p4[i] = Jet_p4[j1] + Jet_p4[j2] + Jet_p4[j3];
-        }
-        return Trijet_p4;
-        """,
+            "Trijet_btag",
+            """
+            auto Jet_btagCSVV2_masked = Jet_btagCSVV2[Jet_pt_mask];
+            auto J1_btagCSVV2 = ROOT::VecOps::Take(Jet_btagCSVV2_masked, Trijet[0]);
+            auto J2_btagCSVV2 = ROOT::VecOps::Take(Jet_btagCSVV2_masked, Trijet[1]);
+            auto J3_btagCSVV2 = ROOT::VecOps::Take(Jet_btagCSVV2_masked, Trijet[2]);
+            return J1_btagCSVV2 > 0.5 || J2_btagCSVV2 > 0.5 || J3_btagCSVV2 > 0.5;
```

Comment:
Does the expression have a nice look? A more natural look would have `max(J1_btagCSVV2, J2_btagCSVV2, J3_btagCSVV2)>0.5`, but I haven't seen such a function in ROOT

Reply:
my brain can easily understand the current expression as "at least one of the elements of J1, J2 and J3 should be greater than the btag threshold". I actually find the version with `max` harder to understand. might be subjective though.

Reply:
Okay, in this case, am removing the corresponding `FIXME`